tutorials/pubsub-spring-dedup-messages/index.md

```diff
@@ -0,0 +1,244 @@
+---
+title: Deduplicate Pub/Sub Messages Using Dataflow in A Spring Boot Application
+description: Use Dataflow's PubsubIO to deduplicate Pub/Sub messages in a Spring Boot application.
+author: anguillanneuf
+tags: PubSub, Spring, Spring Cloud GCP, Dataflow, Java
+date_published: 2020-02-29
+---
+
+
+Tianzi Cai | Developer Programs Engineer | Google Cloud
+
+
+## Objectives
+- Configure a Spring Boot application to use [Cloud Pub/Sub] as a message broker.
+- Use [Cloud Dataflow] to deduplicate messages.
+
+## Before You Begin
+
+1. Install the [Cloud SDK].
+
+1. Create a new Google Cloud project via the
+   [*New Project* page],
+   or via the `gcloud` command line tool.
+
+   ```shell script
+   export PROJECT_NAME=your-google-cloud-project-id
+   gcloud projects create $PROJECT_NAME
+   ```
+
+1. [Enable billing].
+
+1. Setup the Cloud SDK to your GCP project.
+
+   ```shell script
+   gcloud init
+   ```
+
+1. [Enable the APIs](https://console.cloud.google.com/flows/enableapi?apiid=dataflow,compute_component,pubsub): Dataflow, Compute Engine, Pub/Sub.
+
+1. Create a service account JSON key via the
+   [*Create service account key* page],
+   or via the `gcloud` command line tool.
+   Here is how to do it through the *Create service account key* page.
+
+   * From the **Service account** list, select **New service account**.
+   * In the **Service account name** field, enter a name.
+   * From the **Role** list, select **Project > Owner**.
+   * Click **Create**. A JSON file that contains your key downloads to your computer.
```

Comment:
I think the phrasing is a bit strange. Prefer something like "Save this JSON file to a location on your computer".